internal/requests/auth_code_request.go

```diff
@@ -19,33 +21,30 @@ const (
 // AuthCodeRequest stores the values required to request a token from the authority using an authorization code
 type AuthCodeRequest struct {
 	webRequestManager WebRequestManager
-	authParameters    *msalbase.AuthParametersInternal
+	authParameters    msalbase.AuthParametersInternal
 	Code              string
 	CodeChallenge     string
-	ClientCredential  *msalbase.ClientCredential
+	ClientCredential  msalbase.ClientCredential
 	RequestType       AuthCodeRequestType
 }
 
 // CreateAuthCodeRequest creates an instance of AuthCodeRequest
-func CreateAuthCodeRequest(
-	webRequestManager WebRequestManager,
-	authParameters *msalbase.AuthParametersInternal,
-	reqType AuthCodeRequestType) *AuthCodeRequest {
-	req := &AuthCodeRequest{
+func CreateAuthCodeRequest(webRequestManager WebRequestManager, authParameters msalbase.AuthParametersInternal, reqType AuthCodeRequestType) *AuthCodeRequest {
```

Comment:
Did this one need to stay a pointer?

Reply:
Yes, or at least for the moment.

The Execute() here modifies its internal state.  Most *Requests seem to do that.  We could probably still avoid a heap allocations by using pointer methods and not pointer objects.  But that only works if you don't have to use the object in an interface{} and the object isn't passed around in parents by pointer.  

I haven't analyzed the code that deep in order to see if it can be removed.